From cdaf2aa7088a232ff8c8bac08ab26f6b16b31ee5 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:55:50 -0400
Subject: [scsi] qla2xxx: Correctly print out/in mailbox registers

Message-id: <1408391817-6389-3-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87903
O-Subject: [PATCH RHEL 7.1 e-stor 02/69] qla2xxx: Correctly print out/in mailbox registers.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Joe Carnuccio <joe.carnuccio@qlogic.com>

scsi commit id 0e31a2c830a95f7b92f6148d3b4c1f5520978951

At mailbox/buffer debug level, print the correct values of the
outgoing and incoming mailbox registers.

Signed-off-by: Joe Carnuccio <joe.carnuccio@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mbx.c b/drivers/scsi/qla2xxx/qla_mbx.c
index 3d48ba2..0708c2e 100644
--- a/drivers/scsi/qla2xxx/qla_mbx.c
+++ b/drivers/scsi/qla2xxx/qla_mbx.c
@@ -117,33 +117,25 @@ qla2x00_mailbox_command(scsi_qla_host_t *vha, mbx_cmd_t *mcp)
  command = mcp->mb[0];
  mboxes = mcp->out_mb;
 
+ ql_dbg(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1111,
+     "Mailbox registers (OUT):\n");
  for (cnt = 0; cnt < ha->mbx_count; cnt++) {
   if (IS_QLA2200(ha) && cnt == 8)
    optr =
        (uint16_t __iomem *)MAILBOX_REG(ha, &reg->isp, 8);
-  if (mboxes & BIT_0)
+  if (mboxes & BIT_0) {
+   ql_dbg(ql_dbg_mbx, vha, 0x1112,
+       "mbox[%d]<-0x%04x\n", cnt, *iptr);
    WRT_REG_WORD(optr, *iptr);
+  }
 
   mboxes >>= 1;
   optr++;
   iptr++;
  }
 
- ql_dbg(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1111,
-     "Loaded MBX registers (displayed in bytes) =.\n");
- ql_dump_buffer(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1112,
-     (uint8_t *)mcp->mb, 16);
- ql_dbg(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1113,
-     ".\n");
- ql_dump_buffer(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1114,
-     ((uint8_t *)mcp->mb + 0x10), 16);
- ql_dbg(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1115,
-     ".\n");
- ql_dump_buffer(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1116,
-     ((uint8_t *)mcp->mb + 0x20), 8);
  ql_dbg(ql_dbg_mbx + ql_dbg_buffer, vha, 0x1117,
      "I/O Address = %p.\n", optr);
- ql_dump_regs(ql_dbg_mbx + ql_dbg_buffer, vha, 0x100e);
 
  /* Issue set host interrupt command to send cmd out. */
  ha->flags.mbox_int = 0;
@@ -254,9 +246,15 @@ qla2x00_mailbox_command(scsi_qla_host_t *vha, mbx_cmd_t *mcp)
   iptr2 = mcp->mb;
   iptr = (uint16_t *)&ha->mailbox_out[0];
   mboxes = mcp->in_mb;
+
+  ql_dbg(ql_dbg_mbx, vha, 0x1113,
+      "Mailbox registers (IN):\n");
   for (cnt = 0; cnt < ha->mbx_count; cnt++) {
-   if (mboxes & BIT_0)
+   if (mboxes & BIT_0) {
     *iptr2 = *iptr;
+    ql_dbg(ql_dbg_mbx, vha, 0x1114,
+        "mbox[%d]->0x%04x\n", cnt, *iptr2);
+   }
 
    mboxes >>= 1;
    iptr2++;
-- 
1.7.1